### PR TITLE
Expose all ray-query flags in constants

### PR DESCRIPTION
**Pull request by kvark**

### CODE REVIEW

File: src/front/wgsl/parse/mod.rs [hunk 1]
```diff
@@ -638,14 +638,62 @@ impl Parser {
...
+                ast::Expression::Literal(ast::Literal::Number(Number::U32(1)))
+            }
+            (Token::Word("RAY_QUERY_INTERSECTION_GENERATED"), _) => {
+                let _ = lexer.next();
+                ast::Expression::Literal(ast::Literal::Number(Number::U32(2)))
+            }
+            (Token::Word("RAY_QUERY_INTERSECTION_AABB"), _) => {
+                let _ = lexer.next();
+                ast::Expression::Literal(ast::Literal::Number(Number::U32(4)))
+            }
```
Comment (jimblandy) [inline src/front/wgsl/parse/mod.rs:696]: Which Vulkan flags do these correspond to? I wasn't able to find anything that seems to match this.

**Review by jimblandy — CHANGES_REQUESTED** [review]
Since they're going to appear in the IR, these constants need to be a Rust bitflags in `lib.rs`. Please document which Vulkan flag set each one corresponds to.

File: src/front/wgsl/parse/mod.rs (hunk 1 above)
Comment (kvark) [reply src/front/wgsl/parse/mod.rs:696]: Defined here - https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/KHR/SPV_KHR_ray_query.asciidoc

**Comment by kvark** [comment]
Please take another look!

**Comment by cwfitzgerald** [comment]
Hello, thank you for your PR against Naga!

As part of https://github.com/gfx-rs/wgpu/issues/4231, we have moved development of Naga into the wgpu repository in the Naga subfolder. We have transferred all issues, but we are unable to automatically transfer PRs. 

As such, please recreate your PR against the wgpu repository. We apologize for the inconvenience this causes, but will make contributing to both projects more streamlined going forward.

We are leaving PRs open, but once they are transferred, please close the original Naga PR.

**Comment by kvark** [comment]
Ported as https://github.com/gfx-rs/wgpu/pull/5429

**Pull request closed without merging**